pkg/server/config.go

```diff
@@ -56,7 +64,16 @@ func (c Config) Complete() (*server, error) {
 		return nil, fmt.Errorf("unable to construct a client to connect to the kubelets: %v", err)
 	}
 	nodes := informer.Core().V1().Nodes()
-	scrape := scraper.NewScraper(nodes.Lister(), kubeletClient, c.ScrapeTimeout)
+	ns := strings.TrimSpace(c.NodeSelector)
+	if ns != "" {
+		if nodeSelector, err := labels.Parse(ns); err == nil {
+			labelSelector = nodeSelector
+		}
+		if nodeSelector, err := labels.ParseToRequirements(ns); err == nil {
```

Comment:
I think there are difference between requirements and labelSelector. I expect one is a subset of the other (not an expert), however this is not reflected in the code. 

Please document why we parse `--node-selector` as both selector and a requirement. Test would also be helpful.

Reply:
Using requirements and labelSelector at the same time is really confusing.
I updated it and used requirements uniformly, could you please take a look again?
Maybe be easier to understand after updated